diff --git a/conda/base/context.py b/conda/base/context.py
index 5d8d179..dfa8da4 100644
--- a/conda/base/context.py
+++ b/conda/base/context.py
@@ -119,7 +119,7 @@ def mockable_context_envs_dirs(root_writable, root_prefix, _envs_dirs):
     else:
         fixed_dirs = [
             join("~", ".conda", "envs"),
-            join(root_prefix, "envs"),
+            # join(root_prefix, "envs"),
         ]
     if on_win:
         fixed_dirs.append(join(user_data_dir(APP_NAME, APP_NAME), "envs"))
@@ -665,7 +665,7 @@ class Context(Configuration):
         else:
             cache_dir_name = "pkgs32" if context.force_32bit else "pkgs"
             fixed_dirs = (
-                self.root_prefix,
                 join("~", ".conda"),
+                self.root_prefix,
             )
             if on_win:
