diff --git a/lib/cli.js b/lib/cli.js
index 35ed698..e78fc9f 100755
--- a/lib/cli.js
+++ b/lib/cli.js
@@ -2218,7 +2218,7 @@ var NODE_PACKAGE_JSON = exports.NODE_PACKAGE_JSON = 'package.json';
 var PNP_FILENAME = exports.PNP_FILENAME = '.pnp.js';
 
 var POSIX_GLOBAL_PREFIX = exports.POSIX_GLOBAL_PREFIX = `${process.env.DESTDIR || ''}/usr/local`;
-var FALLBACK_GLOBAL_PREFIX = exports.FALLBACK_GLOBAL_PREFIX = path.join(userHome, '.yarn');
+var FALLBACK_GLOBAL_PREFIX = exports.FALLBACK_GLOBAL_PREFIX = path.join(userHome, '.local');
 
 var META_FOLDER = exports.META_FOLDER = '.yarn-meta';
 var INTEGRITY_FILENAME = exports.INTEGRITY_FILENAME = '.yarn-integrity';
