--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -746,14 +746,14 @@
 	tmake_file="$tmake_file pa/t-netbsd"
 	;;
 i[34567]86-*-darwin*)
-	tmake_file="$tmake_file i386/t-crtpc t-crtfm i386/t-msabi"
+	tmake_file="$tmake_file i386/t-crtpc t-crtfm"
 	tm_file="$tm_file i386/darwin-lib.h"
 	extra_parts="$extra_parts crtprec32.o crtprec64.o crtprec80.o"
 	extra_parts="$extra_parts crtfastmath.o libd10-uwfef.a libheapt_w.a"
 	tmake_file="${tmake_file} i386/t-heap-trampoline"
 	;;
 x86_64-*-darwin*)
-	tmake_file="$tmake_file i386/t-crtpc t-crtfm i386/t-msabi"
+	tmake_file="$tmake_file i386/t-crtpc t-crtfm"
 	tm_file="$tm_file i386/darwin-lib.h"
 	extra_parts="$extra_parts crtprec32.o crtprec64.o crtprec80.o"
 	extra_parts="$extra_parts crtfastmath.o libd10-uwfef.a libheapt_w.a"
