diff --git a/CMakeLists.txt b/CMakeLists.txt
index c3062e2..0ed2c1c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,18 +61,6 @@ else(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
   endif(NOT CMAKE_CXX_FLAGS MATCHES "-Wall")

-  # Use -Wextra for clang and gcc.
-  if(NOT CMAKE_CXX_FLAGS MATCHES "-Wextra")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wextra")
-  endif(NOT CMAKE_CXX_FLAGS MATCHES "-Wextra")
-
-  # Use -Werror for clang only.
-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    if(NOT CMAKE_CXX_FLAGS MATCHES "-Werror")
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-    endif(NOT CMAKE_CXX_FLAGS MATCHES "-Werror")
-  endif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-
   # Disable C++ exceptions.
   string(REGEX REPLACE "-fexceptions" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
