--- a/libgcc/config/darwin-unwind.ver
+++ b/libgcc/config/darwin-unwind.ver
@@ -17,14 +17,3 @@
 __Unwind_Backtrace
 __Unwind_Resume_or_Rethrow
 __Unwind_GetIPInfo
-
-___register_frame
-___register_frame_table
-___register_frame_info
-___register_frame_info_bases
-___register_frame_info_table
-___register_frame_info_table_bases
-
-___deregister_frame
-___deregister_frame_info
-___deregister_frame_info_bases
