diff -u a/languages.toml b/languages.toml
--- a/languages.toml	2023-10-25 15:28:16
+++ b/languages.toml	2023-10-25 15:31:37
@@ -63,9 +63,7 @@
 r = { command = "R", args = ["--no-echo", "-e", "languageserver::run()"] }
 racket = { command = "racket", args = ["-l", "racket-langserver"] }
 regols = { command = "regols" }
-rescript-language-server = { command = "rescript-language-server", args = ["--stdio"] }
 robotframework_ls = { command = "robotframework_ls" }
-serve-d = { command = "serve-d" }
 slint-lsp = { command = "slint-lsp", args = [] }
 solargraph = { command = "solargraph", args = ["stdio"] }
 solc = { command = "solc", args = ["--lsp"] }
@@ -1549,21 +1547,6 @@
 source = { git = "https://github.com/elixir-lang/tree-sitter-iex", rev = "39f20bb51f502e32058684e893c0c0b00bb2332c" }
 
 [[language]]
-name = "rescript"
-scope = "source.rescript"
-injection-regex = "rescript"
-file-types = ["res"]
-roots = ["bsconfig.json"]
-auto-format = true
-comment-token = "//"
-language-servers = [ "rescript-language-server" ]
-indent = { tab-width = 2, unit = "  " }
-
-[[grammar]]
-name = "rescript"
-source = { git = "https://github.com/jaredramirez/tree-sitter-rescript", rev = "65609807c628477f3b94052e7ef895885ac51c3c" }
-
-[[language]]
 name = "erlang"
 scope = "source.erlang"
 injection-regex = "erl(ang)?"
@@ -2279,21 +2262,6 @@
 [[grammar]]
 name = "wast"
 source = { git = "https://github.com/wasm-lsp/tree-sitter-wasm", rev = "2ca28a9f9d709847bf7a3de0942a84e912f59088", subpath = "wast" }
-
-[[language]]
-name = "d"
-scope = "source.d"
-file-types = [ "d", "dd" ]
-roots = []
-comment-token = "//"
-injection-regex = "d"
-indent = { tab-width = 4, unit = "    "}
-language-servers = [ "serve-d" ]
-formatter = { command = "dfmt" }
-
-[[grammar]]
-name = "d"
-source = { git = "https://github.com/gdamore/tree-sitter-d", rev="601c4a1e8310fb2f3c43fa8a923d0d27497f3c04" }
 
 [[language]]
 name = "vhs"
