diff --git a/tool/mkconfig.rb b/tool/mkconfig.rb
index e6d61c5..1517669 100755
--- a/tool/mkconfig.rb
+++ b/tool/mkconfig.rb
@@ -190,6 +190,7 @@ module RbConfig
 ]
 print "  # Ruby installed directory.\n"
 print "  TOPDIR = File.dirname(__FILE__).chomp!(#{relative_archdir.dump})\n"
+print "  KEGDIR = File.expand_path('../../../..', __FILE__)\n"
 print "  # DESTDIR on make install.\n"
 print "  DESTDIR = ", (drive ? "TOPDIR && TOPDIR[/\\A[a-z]:/i] || " : ""), "'' unless defined? DESTDIR\n"
 print <<'ARCH' if universal

