--- a/folly/external/aor/CMakeLists.txt
+++ b/folly/external/aor/CMakeLists.txt
@@ -33,20 +33,6 @@
 )
 
 folly_add_library(
-  NAME memcpy_aarch64-use
-  SRCS
-    memcpy-advsimd.S
-    memcpy-armv8.S
-    memcpy-mops.S
-    memcpy_sve.S
-    memmove-mops.S
-  HEADERS
-    asmdefs.h
-  COMPILE_OPTIONS
-    -DFOLLY_MEMCPY_IS_MEMCPY
-)
-
-folly_add_library(
   NAME memset_aarch64
   SRCS
     memset-advsimd.S
@@ -55,15 +41,4 @@
     asmdefs.h
 )
 
-folly_add_library(
-  NAME memset_aarch64-use
-  SRCS
-    memset-advsimd.S
-    memset-mops.S
-  HEADERS
-    asmdefs.h
-  COMPILE_OPTIONS
-    -DFOLLY_MEMSET_IS_MEMSET
-)
-
 endif()
