diff --git a/lib/rubygems/defaults.rb b/lib/rubygems/defaults.rb
index b186375..de168c4 100644
--- a/lib/rubygems/defaults.rb
+++ b/lib/rubygems/defaults.rb
@@ -137,7 +129,7 @@ def self.config_file
   # The path to standard location of the user's state file.

   def self.state_file
-    @state_file ||= File.join(Gem.state_home, "gem", "last_update_check").tap(&Gem::UNTAINT)
+    @state_file ||= File.join(Gem.state_home, "last_update_check").tap(&Gem::UNTAINT)
   end

   ##
